### PR TITLE
LanguageService: add getSupportedCodeFixes to make it proxyable

### PR DESCRIPTION
**Pull request by ajafff**
Fixes: #28966

I'd highly appreciate some pointers on how to test this change. (@sheetalkamat?)

Is the change to the protocol what @DanielRosenwasser meant in https://github.com/Microsoft/TypeScript/issues/28966#issuecomment-447087185?

### CODE REVIEW

File: src/server/session.ts [hunk 1]
```diff
@@ -1803,7 +1803,11 @@ namespace ts.server {
             }
         }
 
-        private getSupportedCodeFixes(): string[] {
+        private getSupportedCodeFixes(args?: protocol.FileRequestArgs): ReadonlyArray<string> {
+            if (args && args.file) {
+                const {project} = this.getFileAndProject(args);
```
Comment (sheetalkamat) [inline src/server/session.ts]: You should probably get project and file and pass file as well to the LS api.

**Comment by ajafff** [comment]
@sheetalkamat I changed `getSupportedCodeFixes` to take the fileName as parameter, although that means the client would need to call `getSupportedCodeFixes` once for each file instead of once per project.
I'm wondering if this API still makes sense as a means for reducing communication to the service. In addition it would need to be refreshed once a plugin's configuration changed in any way. Wouldn't it be easier to just call `getCodeFixes`?

Deprecating this API would also resolve #28990 

---

Regarding tests: as far as I understand, fourslash only tests `Project` and `LanguageService` but not `Session` where the actual change of this PR is. If Session doesn't have tests, do we even need a test for the new method in LanguageService (given that every method in LanguageService can be intercepted by a plugin)?

**Comment by sheetalkamat** [comment]
@ajafff  for test you can look at #28385 to see how to test this through session. 
I am not very familiar with `getSupportedCodeFixes` api and reasoning behind it. @mjbvz might know more about it.

**Comment by sheetalkamat** [comment]
Closing this since #29051 makes more sense than this change

**Pull request closed without merging**